src/execution/evm/instructions/memory.rs

```diff
@@ -20,57 +20,57 @@ pub(crate) fn num_words(size_in_bytes: usize) -> i64 {
 
 #[inline]
 pub(crate) fn mload(state: &mut ExecutionState) -> Result<(), StatusCode> {
-    let index = state.stack.pop();
+    let index = state.stack().pop();
 
     let region = get_memory_region_u64(state, index, NonZeroUsize::new(32).unwrap())
         .map_err(|_| StatusCode::OutOfGas)?;
 
-    let value = u256_from_slice(&state.memory[region.offset..region.offset + region.size.get()]);
+    let value = u256_from_slice(&state.heap()[region.offset..][..region.size.get()]);
```

Comment:
Key point of bumpalo is to avoid this change: keep separate Memory for each stack, they will just reuse the same allocated space underneath, unbeknownst to EVM interpreter.

Reply:
`EvmSubMemory` is used to enforce relative drop order between heap and stack. Without it, it will be easy for a seemingly innocent change to result in a memory leak, e.g. if stack or heap gets temporarily moved somewhere else.